cachi2/core/package_managers/cargo/main.py

```diff
@@ -22,20 +21,29 @@ def fetch_cargo_source(request: Request) -> RequestOutput:
 
     for package in request.cargo_packages:
         package_dir = request.source_dir.join_within_root(package.path)
-        components.extend(_resolve_cargo_package(package_dir, request.output_dir))
         # cargo allows to specify configuration per-package
         # https://doc.rust-lang.org/cargo/reference/config.html#hierarchical-structure
-        project_files.append(_use_vendored_sources(package_dir))
+        fetched_components, cfg_template = _resolve_cargo_package(package_dir, request.output_dir)
+        components.extend(fetched_components)
+        tail = 'directory = "${output_dir}/deps/cargo"'
+        # absolute path has to be replaced with relative path for sources relocation to work:
+        # (absolute path also happens to be the last line generated by cargo)
+        cfg_template = "\n".join(cfg_template.split("\n")[:-2] + [tail])
```

Comment:
`cfg_template.split("\n")[:-2] + [tail]` this is quite cryptic to be fair even if documented, but like I said above, we can't rely on this, the ordering of the config sections may change or new sections may be generated in the future.

Reply:
To be honest, the original version with a static template and a `TOMLKit.update` was more readable than this new attempt.